integrations/opsgenie/main.py

```diff
@@ -38,9 +41,26 @@ async def fetch_oncall(schedule_id: str) -> dict[str, Any]:
     return schedule_batch
 
 
+async def enrich_team_with_members(
+    opsgenie_client: OpsGenieClient,
+    semaphore: asyncio.Semaphore,
+    team_batch: list[dict[str, Any]],
+) -> list[dict[str, Any]]:
+    async def fetch_team_members(team_id: str) -> dict[str, Any]:
+        async with semaphore:
+            return await opsgenie_client.get_team_members(team_id)
+
+    team_tasks = [fetch_team_members(team["id"]) for team in team_batch]
+    results = await asyncio.gather(*team_tasks)
+
+    for team, members in zip(team_batch, results):
+        team["__members"] = members
+
+    return team_batch
```

Comment:
lets have a semaphore stored for `get_team_members` in the client, that will be used inside of the client.
I don't think the "main" code should handle the semaphore, but rather the client should handle how many requests it supports